### DISCUSSION TITLE
Handling Nested Iterations with x-for in Alpine.js

### DISCUSSION THREAD

**Discussion by farismohammed** [alpinejs/alpine#3626] [1. Help] [answered]
I am encountering challenges when attempting to handle nested iterations using the x-for directive in Alpine.js. I require assistance in understanding and implementing the correct approach to deal with such nested cases.

Here is the scenario:

I have a data structure with nested elements, and I would like to iterate over them and display the content accordingly. However, I am facing difficulties due to the limitations of the x-for directive.

```html
<div x-data="{ students: [
    { title: 'Student 1', children: [{ name: 'John', age: 25 }, { name: 'Jane', age: 30 }] },
    { title: 'Student 2', children: [{ name: 'Bob', age: 40 }, { name: 'Alice', age: 35 }] }
  ] }">
  <table>
    <thead>
      <tr>
        <th colspan="2">Students</th>
      </tr>
    </thead>
    <tbody>
      <template x-for="(student, index) in students" :key="index">
        <tr>
          <td colspan="2" x-text="student.title"></td>
        </tr>
        <template x-for="child in student.children" :key="child.name">
          <tr>
            <td x-text="child.name"></td>
            <td x-text="child.age"></td>
          </tr>
        </template>
      </template>
    </tbody>
  </table>
</div>
```

I kindly request guidance on how to effectively handle the nested iterations using x-for in Alpine.js. Any insights, best practices, or alternative approaches to tackle this issue would be greatly appreciated.

**Comment by SimoTod** [comment, marked as answer]
A table can have multiple tbody and they semantically make sense for your use case.
```html
<div x-data="{ students: [ { title: 'Student 1', children: [{ name: 'John', age: 25 }, { name: 'Jane', age: 30 }] }, { title: 'Student 2', children: [{ name: 'Bob', age: 40 }, { name: 'Alice', age: 35 }] } ] }">
  <table>
    <thead>
      <tr>
        <th colspan="2">Students</th>
      </tr>
    </thead>
    <template x-for="(student, index) in students" :key="index">
      <tbody>
        <tr>
          <td colspan="2" x-text="student.title"></td>
       </tr>
       <template x-for="child in student.children" :key="child.name">
         <tr>
           <td x-text="child.name"></td>
           <td x-text="child.age"></td>
         </tr>
       </template>
     </tbody>
   </template>
 </table>
</div>
```
Alternativly, a simple div with `display: contents` instead of the tbody tag would do.

**Reply by farismohammed** [reply]
Thank you, my friend, for your invaluable help. Your solution has successfully resolved the issue I was facing. I am truly grateful for your assistance.

**Reply by fwta-dev** [reply]
Thank you for your help . But missing end of tag </template>

**Reply by SimoTod** [reply]
> Thank you for your help . But missing end of tag </template> 
> 

Added, thanks

**Reply by gitvinodh** [reply]
@SimoTod Thank you for your help. Extremely useful and stopped me from running around in circles. Thanks again.
